add `points_from_xy` to `GeoSeries` class

tsutterley opened the pull request:
Parallels function of same name in GeoPandas for creating a GeoSeries from in-memory arrays

tsutterley committed:
add `point_from_xy` to `GeoSeries` class

tsutterley committed:
add crs attribute following #64

kylebarron commented on py-geopolars/python/geopolars/internals/geoseries.py:
```diff
@@ -205,6 +205,29 @@ def geoarrow_coords_to_numpy(struct_array: pyarrow.StructArray):
         raise ValueError()
 
+    @classmethod
+    def points_from_xy(cls, x, y,
```
I think you also want the type hints on `x` and `y`? Note that `Iterable | pl.Series` is a strict subset of what you can pass into `pyarrow.StructArray`; it also accepts numpy arrays and pyarrow arrays I think

kylebarron commented:
Thanks for the PR! I'm not exactly sure at a glance why CI is now failing. Regardless I think you might need to apply `black`

tsutterley committed:
review updates and formatting

remove z coordinates (unsupported)
black formatting of `geoseries.py`
add typing for x and y coordinates

tsutterley commented:
Removing z makes things simpler :)
applied `black` formatting so we'll see how it goes with CI
happy to contribute!
